### PR TITLE
virtctl: Apply --namespace to created manifests

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

With this PR virtctl applies the namespace passed with the
--namespace flag to the following created manifests:

- VirtualMachine
- VirtualMachinePreferences
- VirtualMachineInstancetype
- VirtualMachineClone

**Release note**:

```release-note
virtctl: Apply namespace to created manifests
```

### CODE REVIEW

**Review by lyarwood — APPROVED** [review]
/lgtm

File: pkg/virtctl/create/instancetype/instancetype.go [hunk 1]
```diff
@@ -107,16 +113,27 @@ func NewCommand() *cobra.Command {
...
 
-func (c *createInstancetype) setDefaults(cmd *cobra.Command) {
+func (c *createInstancetype) setDefaults(cmd *cobra.Command) error {
+	namespace, overridden, err := c.clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+	if overridden {
+		c.namespace = namespace
+		c.namespaced = true
```
Comment (opokornyy) [inline pkg/virtctl/create/instancetype/instancetype.go:123]: Do we still need the `namespaced` flag? We could by default create `VirtualMachineClusterInstancetype` and based on `namespace` flag decide if we want the `VirtualMachineInstancetype` instead. What do you think about that?

File: pkg/virtctl/create/preference/preference.go [hunk 2]
```diff
@@ -80,16 +85,27 @@ func NewCommand() *cobra.Command {
...
 
-func (c *createPreference) setDefaults(cmd *cobra.Command) {
+func (c *createPreference) setDefaults(cmd *cobra.Command) error {
+	namespace, overridden, err := c.clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+	if overridden {
+		c.namespace = namespace
+		c.namespaced = true
```
Comment (opokornyy) [inline pkg/virtctl/create/preference/preference.go:95]: Same question like for VirtualMachineInstancetype.

File: pkg/virtctl/create/instancetype/instancetype.go (hunk 1 above)
Comment (0xFelix) [reply pkg/virtctl/create/instancetype/instancetype.go:123]: I'd keep the `namespaced` flag so it is still possible to create a `VirtualMachineInstancetype` that does not have the namespace set.

File: pkg/virtctl/create/clone/clone.go [hunk 3]
```diff
@@ -23,14 +23,15 @@ import (
 	"fmt"
 	"strings"
 
-	"kubevirt.io/kubevirt/pkg/pointer"
-
 	"github.com/spf13/cobra"
```
Comment (jcanocan) [inline pkg/virtctl/create/clone/clone.go:26]: it's not related with this PR but, newline?

File: pkg/virtctl/create/create.go [hunk 4]
```diff
@@ -21,9 +21,9 @@ package create
 
 import (
 	"github.com/spf13/cobra"
```
Comment (jcanocan) [inline pkg/virtctl/create/create.go:23]: newline?

**Review by jcanocan — COMMENTED** [review]
Nice job. Looks good to me! Just a couple small comments.

File: pkg/virtctl/create/clone/clone.go (hunk 3 above)
Comment (0xFelix) [reply pkg/virtctl/create/clone/clone.go:26]: I moved this local import to the bottom of the imports on purpose.

File: pkg/virtctl/create/create.go (hunk 4 above)
Comment (0xFelix) [reply pkg/virtctl/create/create.go:23]: Basically the same as the other comment: First stdlib imports, then remote packages, then local packages. I did this on purpose while touching the imports anyways.

File: pkg/virtctl/create/clone/clone.go (hunk 3 above)
Comment (jcanocan) [reply pkg/virtctl/create/clone/clone.go:26]: Sorry, I didn't explain myself properly. I meant between:

Comment (0xFelix) [reply pkg/virtctl/create/clone/clone.go:26]: Those are all remote imports, so AFAIK we don't separate them further.

**Comment by 0xFelix** [comment]
@alicefr Ping :)

**Comment by alicefr** [comment]
It looks good, the only super nit I have, if we should put the common field in a struct instead of adding them to every types. Like:
```golang
type createCommon {
     name string
     namespace string
     clientConfig clientcmd.ClientConfig
}
```
(not sure about the name) and a common function to set the defaults for those values.

**Comment by alicefr** [comment]
/approve
/hold
if you want to address the comments.

**Comment by 0xFelix** [comment]
I have looked at your suggestions and I don't think they necessarily make things any easier. It would require public fields, which I'd like to avoid and a common defaults setter for the namespace is not possible, because they all require slightly different logic.

**Comment by alicefr** [comment]
> I have looked at your suggestions and I don't think they necessarily make things any easier. It would require public fields, which I'd like to avoid and a common defaults setter for the namespace is not possible, because they all require slightly different logic.
> 
> Maybe in a follow up?

That's ok, thanks for your work!
/unhold

**Pull request merged**